botcore/db.py

```diff
@@ -0,0 +1,162 @@
+import firebase_admin
+from firebase_admin import credentials, firestore
+import google.cloud.exceptions
+
+from discord.ext import commands
+
+class MemberNotFound(Exception):
+    pass
+
+class Database(commands.Cog):
+    CERTIFICATE_FILE = "config/firebase_credentials.json"
+    COL_MEMBERS = "members"
+    COL_SECRETS = "secrets"
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.db = None
+        self._connect()
+
+    def get_member_data(self, id):
+        """
+        Retrieve entry for a member from the database.
+
+        Args:
+            id: Discord ID of a member.
+        
+        Returns:
+            Dictionary containing all keys and values associated with a member.
+        
+        Raises:
+            MemberNotFound: If member does not exist in the database.
+        """
+
+        data = self._get_member_doc(id).get().to_dict()
+        if data is None:
+            raise MemberNotFound
+        return data
+
+    def get_unverified_members_data(self):
+        """
+        Retrieve entries for all unverified members from the database.
+
+        Returns:
+            Dictionary where each key is a member ID and each value is the info
+            associated with that member.
+        """
+
+        unverified = {}
+        docs = self._get_members_col() \
+            .where("id_verified", "==", False).stream()
```

Comment:
Comparing boolean values with equality? I'm not sure what the `.where()` function wants as arguments but is this best practice?

Reply:
This is how the Firebase SDK works, which I don't really like either :(